benchmark/Cajita_FastFourierTransformPerformance.cpp

```diff
@@ -0,0 +1,232 @@
+
+    // number of runs in test loops
+    int num_runs = 10;
+
+    // get MPI rank
+    int comm_rank;
+    MPI_Comm_rank( comm, &comm_rank );
+
+    // create timers
+    Cabana::Benchmark::Timer setup_timer( test_prefix + "setup",
+                                          num_grid_size_per_dim );
+
+    Cabana::Benchmark::Timer transforms_timer( test_prefix + "transforms",
+                                               num_grid_size_per_dim );
+    // loop over the grid sizes
+    for ( int p = 0; p < num_grid_size_per_dim; ++p )
+    {
+        double cell_size = 1.0 / grid_sizes_per_dim[p];
+        auto global_mesh = createUniformGlobalMesh(
+            global_low_corner, global_high_corner, cell_size );
+
+        // Create the global grid
+        DimBlockPartitioner<3> partitioner;
+        auto global_grid =
+            createGlobalGrid( comm, global_mesh, is_dim_periodic, partitioner );
+
+        // Create a local grid
+        auto local_grid = createLocalGrid( global_grid, 0 );
+        auto owned_space = local_grid->indexSpace( Own(), Cell(), Local() );
+        auto ghosted_space = local_grid->indexSpace( Ghost(), Cell(), Local() );
+
+        auto vector_layout = createArrayLayout( local_grid, 2, Cell() );
+        auto lhs = createArray<double, Device>( "lhs", vector_layout );
+        auto lhs_view = lhs->view();
+        auto lhs_host =
+            createArray<double, typename decltype( lhs_view )::array_layout,
+                        Kokkos::HostSpace>( "lhs_host", vector_layout );
+        auto lhs_host_view = Kokkos::create_mirror_view( lhs_view );
+        uint64_t seed = global_grid->blockId() +
+                        ( 19383747 % ( global_grid->blockId() + 1 ) );
+        using rnd_type = Kokkos::Random_XorShift64_Pool<Kokkos::HostSpace>;
+        rnd_type pool;
+        pool.init( seed, ghosted_space.size() );
+        for ( int i = owned_space.min( Dim::I ); i < owned_space.max( Dim::I );
+              ++i )
+            for ( int j = owned_space.min( Dim::J );
+                  j < owned_space.max( Dim::J ); ++j )
+                for ( int k = owned_space.min( Dim::K );
+                      k < owned_space.max( Dim::K ); ++k )
+                {
+                    auto rand = pool.get_state( i + j + k );
+                    lhs_host_view( i, j, k, 0 ) =
+                        Kokkos::rand<decltype( rand ), double>::draw( rand, 0.0,
+                                                                      1.0 );
+                    lhs_host_view( i, j, k, 1 ) =
+                        Kokkos::rand<decltype( rand ), double>::draw( rand, 0.0,
+                                                                      1.0 );
+                }
+
+        // Copy to the device
+        Kokkos::deep_copy( lhs_view, lhs_host_view );
+
+        setup_timer.start( p );
+
+        // Create FFT options
+        Experimental::FastFourierTransformParams params;
+        params.setAllToAll( true );
```

Comment:
Aren't these the defaults?

Reply:
Yes, these unnecessary lines are now removed.